pkg/operator/configobserver/apiserver/observe_cors.go

```diff
@@ -15,33 +15,40 @@ var clusterDefaultCORSAllowedOrigins = []string{
 	`//localhost(:|$)`,
 }
 
-// ObserveAdditionalCORSAllowedOrigins observes the additionalCORSAllowedOrigins field
-// of the APIServer resource and sets the corsAllowedOrigins field of observedConfig
-func ObserveAdditionalCORSAllowedOrigins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	return innerObserveAdditionalCORSAllowedOrigins(genericListers, recorder, existingConfig, []string{"corsAllowedOrigins"})
+type corsObserver struct {
+	corsAllowedOriginsPath []string
 }
 
-// ObserveAdditionalCORSAllowedOriginsToArguments observes the additionalCORSAllowedOrigins field
-// of the APIServer resource and sets the cors-allowed-origins field in observedConfig.apiServerArguments
-func ObserveAdditionalCORSAllowedOriginsToArguments(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	return innerObserveAdditionalCORSAllowedOrigins(genericListers, recorder, existingConfig, []string{"apiServerArguments", "cors-allowed-origins"})
+// NewCORSObserver creates a new observer for the additionalCORSAllowedOrigins,
+// which either sets the `corsAllowedOrigins` field or the `apiServerArguments.cors-allowed-origins`
+// depending on the useArguments value
+func NewCORSObserver(useArguments bool, prefix ...string) configobserver.ObserveConfigFunc {
```

Comment:
in the future releases, we simply remove `useArguments` and change the function name.

Reply:
you still need to keep the prefix because we will be observing to separate trees even after the old argument handling is gone